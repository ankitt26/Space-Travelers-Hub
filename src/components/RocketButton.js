import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import styles from '../CSS/Rockets.module.css';
import { reserve } from '../Redux/Rockets/rocketSlice';

const RocketButton = ({ Id, reserved }) => {
  console.log(reserved);
  let buttonText = 'Reserve Rocket';
  if (reserved === true) {
    buttonText = 'Cancel Reservation';
  }
  const dispatch = useDispatch();

  const onclick = (event) => {
    const RId = event.target.id;
    dispatch(reserve(RId));
  };
  return (
    <div>
      <button className={styles.button} id={Id} type="submit" onClick={onclick}>
        {buttonText}
      </button>
    </div>
  );
};

RocketButton.propTypes = {
  Id: PropTypes.number.isRequired,
  reserved: PropTypes.string,
};

RocketButton.defaultProps = {
  reserved: 'false',
};
export default RocketButton;
