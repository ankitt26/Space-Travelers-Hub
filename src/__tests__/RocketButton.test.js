import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import RocketButton from '../components/RocketButton';
import Store from "../Redux/store";
import { MemoryRouter } from "react-router";

it("MyProfile component test", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Provider store={Store}>
          <RocketButton />
        </Provider>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

