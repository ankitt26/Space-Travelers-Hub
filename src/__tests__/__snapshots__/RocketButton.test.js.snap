// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile component test 1`] = `
<div>
  <button
    className="button"
    onClick={[Function]}
    type="submit"
  >
    Reserve Rocket
  </button>
</div>
`;
